Makefile

```diff
@@ -199,46 +230,64 @@ managers: ## Build all managers
 
-$(GO_APIDIFF): $(TOOLS_DIR)/go.mod
-	cd $(TOOLS_DIR) && go build -tags=tools -o $(GO_APIDIFF_BIN) github.com/joelanford/go-apidiff
+$(ENVSUBST): # Build gotestsum from tools folder.
+	GOBIN=$(TOOLS_BIN_DIR) $(GO_INSTALL) ${ENVSUBST_PKG} ${ENVSUBST_BIN} ${ENVSUBST_VER}
 
-$(ENVSUBST): $(TOOLS_DIR)/go.mod
-	cd $(TOOLS_DIR) && go build -tags=tools -o $(ENVSUBST_BIN) github.com/drone/envsubst/v2/cmd/envsubst
+$(YQ): # Build yq from tools folder.
+	GOBIN=$(TOOLS_BIN_DIR) $(GO_INSTALL) ${YQ_PKG} ${YQ_BIN} ${YQ_VER}
 
-$(YQ): $(TOOLS_DIR)/go.mod
-	cd $(TOOLS_DIR) && go build -tags=tools -o $(YQ_BIN) github.com/mikefarah/yq/v4
+$(KUSTOMIZE): # Build kustomize from tools folder.
+	GOBIN=$(TOOLS_BIN_DIR) $(GO_INSTALL) ${KUSTOMIZE_PKG} ${KUSTOMIZE_BIN} ${KUSTOMIZE_VER}
 
-$(KUSTOMIZE): # Download kustomize using hack script into tools folder.
-	hack/ensure-kustomize.sh
+$(CONVERSION_VERIFIER): $(TOOLS_DIR)/go.mod # Build conversion-verifier from tools folder.
+	cd $(TOOLS_DIR); go build -tags=tools -o $(BIN_DIR)/conversion-verifier sigs.k8s.io/cluster-api/hack/tools/conversion-verifier
 
-$(GOLANGCI_LINT): .github/workflows/golangci-lint.yml # Download golanci-lint using hack script into tools folder.
+$(GOLANGCI_LINT): .github/workflows/golangci-lint.yml # Download golangci-lint using hack script into tools folder.
 	hack/ensure-golangci-lint.sh \
 		-b $(TOOLS_DIR)/$(BIN_DIR) \
 		$(shell cat .github/workflows/golangci-lint.yml | grep version | sed 's/.*version: //')
 
-envsubst: $(ENVSUBST) ## Build a local copy of envsubst.
-kustomize: $(KUSTOMIZE) ## Build a local copy of kustomize.
-setup-envtest: $(SETUP_ENVTEST) ## Build a local copy of setup-envtest.
-controller-gen: $(CONTROLLER_GEN) ## Build a local copy of controller-gen.
-conversion-gen: $(CONVERSION_GEN) ## Build a local copy of conversion-gen.
-conversion-verifier: $(CONVERSION_VERIFIER) ## Build a local copy of conversion-verifier.
-gotestsum: $(GOTESTSUM) ## Build a local copy of gotestsum.
-yq: $(YQ)  ## Build a local copy of yq.
+.PHONY: $(ENVSUBST_BIN)
+$(ENVSUBST_BIN): $(ENVSUBST) ## Build a local copy of envsubst.
+
+.PHONY: $(KUSTOMIZE_BIN)
+$(KUSTOMIZE_BIN): $(KUSTOMIZE) ## Build a local copy of kustomize.
+
+.PHONY: $(SETUP_ENVTEST_BIN)
+$(SETUP_ENVTEST_BIN): $(SETUP_ENVTEST) ## Build a local copy of setup-envtest.
+
+.PHONY: $(CONTROLLER_GEN_BIN)
+$(CONTROLLER_GEN_BIN): $(CONTROLLER_GEN) ## Build a local copy of controller-gen.
+
+.PHONY: $(CONVERSION_GEN_BIN)
+$(CONVERSION_GEN_BIN): $(CONVERSION_GEN) ## Build a local copy of conversion-gen.
+
+.PHONY: $(GOTESTSUM_BIN)
+$(GOTESTSUM_BIN): $(GOTESTSUM) ## Build a local copy of gotestsum.
+
+.PHONY: $(YQ_BIN)
+$(YQ_BIN): $(YQ)  ## Build a local copy of yq.
+
+.PHONY: $(CONVERSION_VERIFIER_BIN)
+$(CONVERSION_VERIFIER_BIN): $(CONVERSION_VERIFIER) ## Build a local copy of conversion-verifier.
+
+.PHONY: $(GO_APIDIFF_BIN)
+$(GO_APIDIFF_BIN): $(GO_APIDIFF) ## Build a local copy of go-apidiff
```

Comment:
Do we need these targets here, or can we dedupe?

Reply:
we need these to be able to call "make binary" in scripts etc. without needing the versioned binary or the full path of the file. Because these in turn call the versioned binary file task, it ensures it only installs it if the version installed doesn't match the desired version or if the binary is missing. For example, if you call `make go-apidiff`, this target will call `/Users/vince/go/src/sigs.k8s.io/cluster-api/hack/tools/bin/go-apidiff-v0.1.0` which means it will install it if you previously had `v0.0.3`, but that way you can always call the same make target in your scripts.

Reply:
Sounds good, I was mostly wondering if we can dedupe them with the other targets